lib/ecto/type.ex

```diff
@@ -404,23 +404,14 @@ defmodule Ecto.Type do
   defp dump_naive_datetime_usec(%NaiveDateTime{microsecond: {_, 6}} = term), do: {:ok, term}
   defp dump_naive_datetime_usec(_), do: :error
 
-  defp dump_utc_datetime(%DateTime{time_zone: time_zone, microsecond: {0, 0}} = term) do
-    if time_zone != "Etc/UTC" do
-      message = ":utc_datetime expects the time zone to be \"Etc/UTC\", got `#{inspect(term)}`"
-      raise ArgumentError, message
-    end
+  defp dump_utc_datetime(%DateTime{microsecond: {0, 0}} = datetime),
+    do: {:ok, datetime |> check_utc_timezone!(:utc_datetime)}
```

Comment:
perhaps we don't need to pipe since it's only one statement?

```suggestion
    do: {:ok, check_utc_timezone!(datetime, :utc_datetime)}
```

Reply:
let's leave it as is, I'll add another check here in a separate PR.